auth/providers/azure/azure.go

```diff
@@ -104,6 +105,8 @@ func New(ctx context.Context, opts Options) (auth.Interface, error) {
 	switch opts.AuthMode {
 	case ClientCredentialAuthMode:
 		c.graphClient, err = graph.New(c.ClientID, c.ClientSecret, c.TenantID, c.UseGroupUID, authInfoVal.AADEndpoint, authInfoVal.MSGraphHost)
+	case ARCAuthMode:
+		c.graphClient, err = graph.NewWithARC(c.MSIAudience)
```

Comment:
we should pass `clientID` in new client such that we have client id validation in token.

Reply:
done